src/settings/window_geometry.rs

```diff
@@ -1,16 +1,63 @@
-#[derive(Debug, Clone)]
+use crate::renderer::Renderer;
+use crate::settings::SETTINGS;
+use crate::window::WindowSettings;
+use glutin::dpi::PhysicalSize;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub struct WindowGeometry {
     pub width: u64,
     pub height: u64,
 }
 
+#[cfg(unix)]
+const SETTINGS_PATH_EXTENSION: &str = ".local/share/nvim/neovide-settings.json";
+#[cfg(windows)]
+const SETTINGS_PATH_EXTENSION: &str = "AppData/Local/nvim-data/neovide-settings.json";
```

Comment:
Should we use something like the dirs package https://docs.rs/dirs/3.0.2/dirs/fn.data_dir.html? It adds a dependency (whats another one right 😂) but at least this way we don't have to have platform specific strings...

Reply:
An a related point, why for unix is it nvim when on windows its nvim-data?

Reply:
![nvim_datapath](https://user-images.githubusercontent.com/28494324/125827724-fb164687-45a4-4772-b9b3-698309a8a85b.png)

According to the nvim help manual, that's what they chose the paths to be lol.

Look like using the dirs package will produce a separate save location compared to the manual. But like I said in the previous pull request, it's just a soft dependency, so I'm fine with either method.

Reply:
Welp looks like theres not really a better way to do this, so just merging as is